torch/csrc/jit/script/module.h

```diff
@@ -0,0 +1,246 @@
+#pragma once
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/graph_executor.h"
+#include "torch/csrc/autograd/variable.h"
+
+namespace torch { namespace jit { namespace script {
+
+struct Method {
+  Method(std::string name, bool optimize)
+  : name_(std::move(name))
+  , graph_(std::make_shared<Graph>())
+  , optimize(optimize) {}
+  variable_tensor_list run(variable_tensor_list && inputs) {
```

Comment:
How does this work? If there are method calls to other methods in a module, I don't see the GraphExecutor being told how to dispatch those calls.

Reply:
Ah, the point is you're supposed to call `emit_call_to` for any such method call, so that there is never actually a method call inside this Method.

Reply:
Right -- when we get around to actual method calls, this will change. Because we expect this to change, I've tried to keep the details of method invocation _inside_ the method class, which is why `emit_call_to` is here rather than in compiler.cpp